src/browser/BrowserAction.h

```diff
@@ -87,13 +87,17 @@ class BrowserAction : public QObject
     QByteArray base64Decode(const QString& str);
     QString incrementNonce(const QString& nonce);
 
+    void setKeys(const QString& publicKey, const QString& secretKey, const QString& clientPublicKey);
```

Comment:
I would actually prefer the test class directly set the private variables in this case since you are making it a friend. Otherwise recommend naming this function "testSetKeys" or similar.

Reply:
Fixed.